src/hooks/useInput.ts

```diff
@@ -0,0 +1,24 @@
+import * as React from "react"
+
+export const useInput = (validateConditionFuntion: (arg: string) => boolean) => {
+  const [value, setValue] = React.useState("")
+  const [isTouched, setIsTouched] = React.useState(false)
+
+  const isValueValid = validateConditionFuntion(value)
+
+  const hasError = !isValueValid && isTouched
+
+  const valueChangeHanlder = (event: any) => {
+    setValue(event.target.value)
+  }
+
+  const inputBlurHandler = (event: any) => {
+    setIsTouched(true)
+  }
+  const reset = () => {
+    setValue("")
+    setIsTouched(false)
+  }
+  return { value: value, isValueValid: isValueValid, hasError: hasError, valueChangeHanlder, inputBlurHandler, reset }
```

Comment:
tại sao value, isValueValid, hasError e ko làm như valueChangeHanlder, inputBlurHandler, reset?
```suggestion
  return { value, isValueValid, hasError, valueChangeHanlder, inputBlurHandler, reset }
```